server/src/Application/Validation/Validators/Lecture/LectureValidationExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using FluentValidation;
+using StudyZen.Application.Services;
+using StudyZen.Domain.Constraints;
+
+namespace StudyZen.Application.Validation;
+
+public static class LectureValidationExtensions
+{
+    public static IRuleBuilderOptions<T, string?> LectureName<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+            .NotEmpty()
+            .WithErrorCode(ValidationErrorCodes.MustNotBeEmpty)
+            .MinimumLength(LectureConstraints.NameMinLength)
+            .WithErrorCode(ValidationErrorCodes.TooShort)
+            .MaximumLength(LectureConstraints.NameMaxLength)
+            .WithErrorCode(ValidationErrorCodes.TooLong);
```

Comment:
Aren't lecture and course rules for the name and description identical? Why not merge them into one? I get it might be useful if we want to add different rules to each later, but i don't think it that will be the case in our project.

Reply:
Removed minimum length requirement since it does not really make sense from database point. DB only cares about max length, not min length.  Also extracted rules to extension methods to make them reusable.